clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -923,6 +965,7 @@ private void revokeAndAssign(SortedSet<TopicIdPartition> assignedTopicIdPartitio
             } else {
                 boolean memberHasRejoined = memberEpochOnReconciliationStart != memberEpoch;
                 if (state == MemberState.RECONCILING && !memberHasRejoined) {
+                    metricsManager.recordRebalanceEnded(time.milliseconds());
```

Comment:
I think there is at least one more situation where we finish a rebalance, which is when we transition from `RECONCILING` to `STABLE` directly when an unreconciled partition is removed from the assignment by the broker. In general, wouldn't it be simpler to just handle this inside `transititionTo`, start the measurement when we enter `RECONCILING` and stop the measurement when we leave `RECONCILING`?